source/dubregistry/api.d

```diff
@@ -46,4 +46,45 @@ class DubRegistryWebApi {
 		if (stats.type != Json.Type.null_) res.writeJsonBody(stats);
 		else res.writeJsonBody(["message": "Package/Version not found"], HTTPStatus.notFound);
 	}
+
+    // searches for a package
+    void querySearch(HTTPServerRequest req, HTTPServerResponse res){
+        string* q = "q" in req.query;
+        
+        if(q == null) 
+        {
+            res.statusCode = 400;
+            res.writeJsonBody("Error: Must pass query parameter");
+            logDiagnostic("Error: no query in search string");
+            return;
+        }
+
+        string[] queryStrings = (*q).split(",");
+
+        if(queryStrings.length == 0)
```

Comment:
You can elegantly handle both errors in one case using [get](http://vibed.org/api/vibe.utils.dictionarylist/DictionaryList.get).
```d
auto queries = req.query.get("q", "").splitter(",");
if (queries.empty)
```
Using [splitter](http://dlang.org/phobos/std_array.html#.splitter) instead of split avoids an allocation.